### PR TITLE
refactor(api-gen): show a Github link to the source for API pages

### PR DESCRIPTION
**Pull request by JeanMeche**
Related to angular/angular#55201 & angular/angular#52641

### CODE REVIEW

**Review by twerske — APPROVED** [review]

**Comment by JeanMeche** [comment]
Note: Tests are failing due to the dependency on the extractor provided by the FW repo and updated here: #55201

File: bazel/api-gen/rendering/templates/header-api.tsx [hunk 1]
```diff
@@ -21,6 +21,14 @@ import {DocsPillRow} from './docs-pill-row';
 export function HeaderApi(props: {entry: DocEntryRenderable; showFullDescription?: boolean}) {
   const entry = props.entry;
 
+  // TODO: This link point to the main branch.
+  // When ADEV is not deployed on the main branch branch anymore,
+  // We should update it to point to the tag of the released version which ADEV runs on.
+
+  const sourceUrl = isDocEntryWithSourceInfo(entry)
+    ? `https://github.com/angular/angular/blob/main${entry.source.filePath}#L${entry.source.startLine}-${entry.source.endLine}`
```
Comment (josephperrott) [inline bazel/api-gen/rendering/templates/header-api.tsx:29]: is `entry.source.filePath` guaranteed to be `/` prefixed?

**Review by josephperrott — APPROVED** [review]
LGTM

Just one small question

File: bazel/api-gen/rendering/templates/header-api.tsx (hunk 1 above)
Comment (josephperrott) [reply bazel/api-gen/rendering/templates/header-api.tsx:29]: Looking again closer, it looks like the extraction test json file doesn't have it as prefixed, so it wouldn't even point to the correct place there.

Comment (JeanMeche) [reply bazel/api-gen/rendering/templates/header-api.tsx:29]: I just checked against the FW directly, the change is correct, the path has a leading `/`. I changed the test data to reflect that.

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit db01d538ad135735bc3e0692a8019921b91dc89b.

**Pull request closed without merging**